nmma/joint/maximum_mass_constraint_utils.py

```diff
@@ -0,0 +1,186 @@
+import numpy as np
+import scipy.stats
+import scipy.interpolate
+import scipy.integrate
+
+from bilby.gw.prior import PriorDict
+from bilby.core.prior import Uniform
+import bilby.gw.conversion as conversion
+
+import os
+import sys
+import contextlib
+
+def fileno(file_or_fd):
+    fd = getattr(file_or_fd, 'fileno', lambda: file_or_fd)()
+    if not isinstance(fd, int):
+        raise ValueError("Expected a file (`.fileno()`) or a file descriptor")
+    return fd
+
+@contextlib.contextmanager
+def stdout_redirected(to=os.devnull, stdout=None):
+    """
+    https://stackoverflow.com/a/22434262/190597 (J.F. Sebastian)
+    """
+    if stdout is None:
+       stdout = sys.stdout
+
+    stdout_fd = fileno(stdout)
+    # copy stdout_fd before it is overwritten
+    #NOTE: `copied` is inheritable on Windows when duplicating a standard stream
+    with os.fdopen(os.dup(stdout_fd), 'wb') as copied: 
+        stdout.flush()  # flush library buffers that dup2 knows nothing about
+        try:
+            os.dup2(fileno(to), stdout_fd)  # $ exec >&to
+        except ValueError:  # filename
+            with open(to, 'wb') as to_file:
+                os.dup2(to_file.fileno(), stdout_fd)  # $ exec > to
+        try:
+            yield stdout # allow code to be run with the redirected stdout
+        finally:
+            # restore stdout to its previous value
+            #NOTE: dup2 makes stdout_fd inheritable unconditionally
+            stdout.flush()
+            os.dup2(copied.fileno(), stdout_fd)  # $ exec >&copied
+
+
+
+def baryonic_mass(gravitational_mass, EOS, eos_path_macro, eos_path_micro):
+
+    Msun_to_MeV = 1.11549833*10**60
+    G = 1.4766696910334391 #G/c^2 in km/Msun
+    R, M, L, P0 = np.loadtxt(eos_path_macro+f"/{EOS}.dat", unpack= True, skiprows = 0)
+    N, EPS, P, CS2 = np.loadtxt(eos_path_micro+f"/{EOS}.dat", unpack = True, skiprows = 0)
+    eps_of_p = scipy.interpolate.interp1d(P, EPS, kind = 'linear', bounds_error = False, fill_value =0)
+
+
+    def TOVeq(y, x):
+      p, m = y
+      eps = eps_of_p(p)
+
+      Dp = -G*m*eps/x**2 * (1+p/eps) * (1+4*np.pi*(x**3* p)*8.964603290800086e-07/m) * (1-2*G*m/x)**(-1)
+      Dm = (4*np.pi*x**2*eps)*8.964603290800086e-07 #1 MeV/fm^3 km^2 are 8.965123536*10**(-7) Msun/km
+
+      return [Dp, Dm]
+
+    dr =0.001
+    r = np.interp(gravitational_mass, M, R)
+    p0= np.interp(gravitational_mass, M, P0)
+    eps0 = eps_of_p(p0)
+    m0 = (eps0*4*np.pi/3*dr**3)*8.964603290800086e-07 #convert Mev/fm^3 * km^3 in Msun
+    x = np.arange(dr, r+dr, dr)
+    y0 = [p0, m0]
+
+    with stdout_redirected():
+         p_solv, m_solv = scipy.integrate.odeint(TOVeq, y0=y0, t = x).T
+    n_solv = np.interp(p_solv, P, N)
+
```

Comment:
there is already a tov solver at `nmma/eos/tov.py`, could u take a look into that